File: app/src/main/java/com/jsoft/jcomic/helper/DownloadItemDTO.java
package com.jsoft.jcomic.helper;

public class DownloadItemDTO {
    public BookDTO book;
    public EpisodeDTO episode;

    public DownloadItemDTO(BookDTO book, EpisodeDTO episode) {
        this.book = book;
        this.episode = episode;
        for (EpisodeDTO e: book.getEpisodes()) {
            if (e.getEpisodeUrl().equals(episode.getEpisodeUrl())) {
                e.setImageUrl(episode.getImageUrl());
            }
        }
    }

    public int getEpisodeIndex() {
        for (int i=0;i<book.getEpisodes().size(); i++) {
            if (book.getEpisodes().get(i).getEpisodeUrl().equals(episode.getEpisodeUrl())) {
                return i;
            }
        }
        return -1;
    }

}

File: app/src/main/java/com/jsoft/jcomic/helper/DownloadItemDTO.kt
package com.jsoft.jcomic.helper

class DownloadItemDTO(var book: BookDTO, var episode: EpisodeDTO) {

    val episodeIndex: Int
        get() {
            for (i in 0 until book.episodes.size) {
                if (book.episodes[i].episodeUrl == episode.episodeUrl) {
                    return i
                }
            }
            return -1
        }

    init {
        for (e in book.episodes) {
            if (e.episodeUrl == episode.episodeUrl) {
                e.imageUrl = episode.imageUrl
            }
        }
    }

}
